webapp/README.md

```diff
@@ -1,40 +1,74 @@
-# Web app GUI
+# CUAHSI Jupyter Sync Frontend
 
-This app is a React web app written in TypeScript and styled using SCSS.
+The frontend is a [React](https://reactjs.org/) app written in [TypeScript](https://typescriptlang.org/).
+It uses [Redux](https://redux.js.org/) for state management and gets all of its data from the backend via HTTP requests.
 
-# Setup
+The file `webapp/dist/bundle.js` is a compiled version of the webapp, allowing it to be run without setting up a
+development environment. This file is typically served by the backend.
+
+# Development
 
 Running this code requires [Node.js](https://nodejs.org/en/) v8.0 or above. (We recommend using
-[n](https://github.com/tj/n) for Node version management.)
+[n](https://github.com/tj/n) to install and manage Node versions.) You'll also need a package manager
+(we recommend [Yarn](https://classic.yarnpkg.com/en/docs/install/#debian-stable)).
 
-Once you have installed Node, run the following:
+Once you have Node and Yarn installed, install the web app dependencies:
 
 ```bash
-cd webapp
-yarn install
+$ cd webapp
+$ yarn install
 ```
 
-This will install all of the Node packages needed to build the app.
+Then run the following to start the app up in development mode (with source maps and hot reloading):
+
+```bash
+$ yarn start
+```
 
-# Running
+Then open [http://localhost:3000](http://localhost:3000) to view it in the browser.
 
-## Development
+# Deployment
 
-To run the code in a development mode (with hot reloading and source maps), run:
+If you would like to update the bundled app served by the backend when running in development mode, run
 
-### `yarn start`
+```bash
+$ yarn build
+```
 
-Then open [http://localhost:3000](http://localhost:3000) to view it in the browser.
+To create a new release (i.e. bundled app served by the Jupyter server extension), run
+
+```bash
+$ yarn deploy
+```
+
+This will create a new `webapp/dist/bundle.js`.
+
+# Architecture
+
+The file `src/index.tsx` is the entry point into the app. This file then displays the correct components depending on
+the URL (using [React Router](https://reacttraining.com/react-router/) bound to Redux using
+ [connected-react-router](https://github.com/supasate/connected-react-router)).
+ 
+ ## Pages
+
+The [React components](https://reactjs.org/docs/components-and-props.html) defined in `src/pages` are the various
+"pages" of the app. There are currently only two:
 
-## Production
+* `MainPage.tsx` displays the list of resources
+* `ResourcePage.tsx` displays the resource metadata, file manager, and readme
 
-Running
+## Non-Page Components
 
-### `yarn build`
+The rest of the components are defined in `src/components`. They get their information passed via props from Redux
```

Comment:
maybe describe what a component is? like give an example?

Reply:
I just added a link to the docs on components here as well.